stringbuilder.go

```diff
@@ -3,11 +3,13 @@ package Text
 import (
 	"fmt"
 	"strconv"
+	"sync"
 )
 
 type StringBuilder struct {
```

Comment:
We should add a documentation stating that the stringbuilder is thread safe

Reply:
Added documentation on struct as well as updated README.md to convey thread-safety for mutation operations.